### DISCUSSION TITLE
Azure Fileshare authentication

### DISCUSSION THREAD

**Discussion by chdittmer** [stryker-mutator/stryker-net#3125] [Q&A] [answered]
tldr; Can someone point me to a more detailed guide on how to create the specific type of Azure file storage for Stryker? i.e. Explain it to me like I am five.

When running with baseline enabled I get the following error "Problem authenticating with azure file share. Make sure your SAS is valid."

I am sure that the file share and SAS is valid, if I combine the URL and SAS  and paste it in a browser window I can browse to the file share so I am pretty sure I am setting it up incorrectly for what Stryker needs but I cannot for the life of me figure out what I am doing wrong. 

I added a baseline file that was created locally to the expected folder location to make sure that is not the issue. The URL looks something like this (important parts redacted) and I can navigate to that. 
https://XXXXXXXXX.file.core.windows.net/platespace/StrykerOutput/{ProjectName}/baseline/main/stryker-report.json??sp=racwdli&st=2024-11-22T09:35:11Z&se=2025-07-01T17:35:11Z&spr=https&sv=2022-11-02&sr=c&sig=XXXXXXXXXXX

**Comment by rouke-broersma** [comment]
Hi, did you follow the instructions in https://stryker-mutator.io/docs/stryker-net/configuration/#azure-fileshare-sas-string ?

**Comment by chdittmer** [comment, marked as answer]
I followed the instructions to the best of my ability but what got me was the ability in azure to create SAS tokens in multiple ways and each has a different scope. I did manage to resolve my issue by generating the SAS in the right place it just took some fiddling. The SAS needs to be generated for the entire storage account and not just for the individual file share. It is also confusing because Azure has a new service in Preview that is called "File shares" that is something completely different.

If anyone runs into this looking for the resolution you should:

- Create a storage account
- Create a file share for that account 
- Create a SAS with the parameters as in the documentation

**Reply by rouke-broersma** [reply]
If you would be willing to provide more accurate instructions in our docs that would be much appreciated. The issue with creating the sas for the file share only is that it doesn't give read permission on the storage account, which I think our current client requires to build the file share uri.
